curl_cffi/requests/session.py

```diff
@@ -1110,7 +1111,10 @@ async def perform():
                     await q.put(STREAM_END)  # type: ignore
 
             def cleanup(fut):
-                self.release_curl(curl)
+                nonlocal curl_released
+                if not curl_released:
+                    self.release_curl(curl)
+                curl_released = True
```

Comment:
Random question: Should the releasing be protected by a [lock](https://docs.python.org/3/library/threading.html#threading.Lock)?